.github/workflows/publish-book.yml

```diff
@@ -0,0 +1,21 @@
+name: publish-book
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: build
+        run: nix develop --print-build-logs --no-update-lock-file .#book-dev --command mdbook build book
+      - name: deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./book/book
+          destination_dir: ./
+          cname: nix.fuel.network
```

Comment:
@Voxelot do you know what we might need to do with the `fuel.network` hosting to get this `cname` to work like we have for [rust.fuel.network](https://github.com/FuelLabs/fuels-rs/blob/002bc8f123e749d77e6681ff35ed4fba480ede59/.github/workflows/gh-pages.yml#L29)?